### DISCUSSION TITLE
Parallel edges and edge color

### DISCUSSION THREAD

**Discussion by flbfilhoUSP** [gephi/gephi#2766] [Q&A]
1. What is the best way to treat parallel edges of different types/label? Currently, the weight is not adding up correctly.

2. Is there a way to color edges according to a type/label? Currently, I'm using a plugin (Give Color to Edges) but it's not very good (does not considers the weights).

**Comment by mbastian** [comment]
Hi @flbfilhoUSP ,

1. Our definition of parallel edges is when they have the same type (also sometimes called 'kind'). So if you have edge type A and edge type B between 2 nodes, we don't consider those parallel edges. It depends a bit how you're importing your data into Gephi but when parallel edges are detected at import time, the report panel proposes you to pick a "merge strategy". There, you have a "sum" merge strategy that would sum the edge weight.

2. You should be able to use the Appearance module to color your edges based on their type. That said, if you want to have a different gradient for each type that is not something we currently support out of the box in Appearance. To do that, you would need to filter the graph by Edge type to keep one edge type at the time and then use the Appearance "Ranking" section. There, you should have the "local scale" option enabled to make sure the scale is only based on that type's weight.

**Reply by flbfilhoUSP** [reply]
Thank you very much for your reply.

I'm new to Gephi but keep learning new stuff every day. The program is really intuitive.

1. Yes, summing is perfect for my case but I'm experiencing exactly the problem you described as not happening. For example, in my data  I have 
- 30 rows labeled A from node X to node Y (or vice versa - undirected);
- 2 rows labeled B from node X to node Y (or vice versa - undirected). 
When I import it, the sum is 32 edges labeled B from node X to Y. Of course, the desired result would be:
- 30 edges labeled A from node X to node Y;
- 2 edges labeled B from node X to node Y. 

2. Because of the problem described before, it doesn't work when I color the edges. I didn't expect it to color in a gradient but somehow show edges A and B from node X to node Y simultaneously. Also, I tried using filters but, since it seems I'm having an import problem, couldn't make it work either.

Again, thank you very much for your attention. Sorry if I'm asking for redundant or not supported stuff. I still have lots to learn but Gephi is already helping me a lot.
